### PR TITLE
tool(platform): Add storybooks to aid UI development 

### PR DESCRIPTION
**Pull request by Swiftyos**
### Background

Storybook is a frontend workshop for building UI components and pages in isolation. It's used for UI development, testing, and documentation.

This PR adds storybook to the autogpt platform frontend to aid in development of our frontend

### Changes 🏗️

- Add storybooks to project
- Add stories to all of the `src/components/ui` components

Only for the new autogpt platform, currently in autogpt_platform/

- Create from scratch and execute an agent with at least 3 blocks
- Import an agent from file upload, and confirm it executes correctly
- Upload agent to marketplace
- Import an agent from marketplace and confirm it executes correctly
- Edit an agent from monitor, and confirm it executes correctly

### CODE REVIEW

File: autogpt_platform/frontend/src/stories/Button.stories.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import type { Meta, StoryObj } from "@storybook/react";
```
Comment (Swiftyos) [inline autogpt_platform/frontend/src/stories/Button.stories.ts:1]: I've left the sample content in for now as it includes docs and how to create a page, which will come in useful during the initial using storybooks

File: autogpt_platform/frontend/src/components/ui/button.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
+import type { Meta, StoryObj } from "@storybook/react";
```
Comment (Swiftyos) [inline autogpt_platform/frontend/src/components/ui/button.stories.tsx:1]: This is the story I'd been working on as an example of the various features of storybook

File: autogpt_platform/frontend/src/stories/Button.tsx
Comment (Pwuts) [inline autogpt_platform/frontend/src/stories/Button.tsx:1]: isn't this a duplicate of `components/ui/button`?

File: autogpt_platform/frontend/src/components/ui/command.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  },
+  tags: ["autodocs"],
+} satisfies Meta<typeof Command>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  render: () => (
+    <Command className="rounded-lg border shadow-md">
```
Comment (Pwuts) [inline autogpt_platform/frontend/src/components/ui/command.stories.tsx]: What are these overrides for?

File: autogpt_platform/frontend/.gitignore [hunk 4]
```diff
@@ -42,3 +42,6 @@ node_modules/
 /playwright-report/
 /blob-report/
 /playwright/.cache/
+
+*storybook.log
+storybook-static
```
Comment (Pwuts) [inline autogpt_platform/frontend/.gitignore]: missing newline

File: autogpt_platform/frontend/src/components/ui/checkbox.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    checked: true,
+  },
+};
+
+export const WithLabel: Story = {
+  args: {},
+  render: (args) => (
+    <div className="flex items-center space-x-2">
+      <Checkbox id="terms" {...args} />
+      <label
```
Comment (Pwuts) [inline autogpt_platform/frontend/src/components/ui/checkbox.stories.tsx:59]: we have a `Label` component

File: autogpt_platform/frontend/src/stories/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
```
Comment (ntindle) [inline autogpt_platform/frontend/src/stories/Header.tsx:1]: Maybe the main file here should be in a folder alongside page

**Comment by ntindle** [comment]
Glad we will have this going forward especially as we build a cohesive design

File: autogpt_platform/frontend/src/stories/Button.tsx
Comment (Swiftyos) [reply autogpt_platform/frontend/src/stories/Button.tsx:1]: Yes please note my previous comment about leaving the default storybook content in as guidance for us during this initial phase of using storybook

File: autogpt_platform/frontend/src/stories/Header.tsx (hunk 6 above)
Comment (Swiftyos) [reply autogpt_platform/frontend/src/stories/Header.tsx:1]: Please note my comment about leaving the default storybook content in as guidance during this initial phase of using storybook.

Once we are all familiar and have completed any advanced setup then we can delete these.

File: autogpt_platform/frontend/src/components/ui/checkbox.stories.tsx (hunk 5 above)
Comment (Swiftyos) [reply autogpt_platform/frontend/src/components/ui/checkbox.stories.tsx:59]: This is testing the checkbox component, it's best to use plain HTML for anything else in the story. Just like normal tests you don't want to be needing to pull in other classes that you are not testing

File: autogpt_platform/frontend/src/components/ui/command.stories.tsx (hunk 3 above)
Comment (Swiftyos) [reply autogpt_platform/frontend/src/components/ui/command.stories.tsx]: You're right these shouldn't be there, I'll remove them. Thanks

File: autogpt_platform/frontend/.gitignore (hunk 4 above)
Comment (Swiftyos) [reply autogpt_platform/frontend/.gitignore]: done

**Review by ntindle — CHANGES_REQUESTED** [review]
The fonts don't seem to be loading. Not sure why as I only looked at this tutorial https://stackonfire.com/posts/shadcn-storybook-visual-testing/ but here's a photo for reference
image

**Comment by majdyz** [comment]
Is there a note on how to use it in our project?
A readme or some notes on the PR for now will do.

**Comment by Swiftyos** [comment]
> The fonts don't seem to be loading. Not sure why as I only looked at this tutorial https://stackonfire.com/posts/shadcn-storybook-visual-testing/ but here's a photo for reference image

Font Family Times is the default font for the project so, whilst it does look off it is working as expected.

**Comment by Swiftyos** [comment]
> Is there a note on how to use it in our project? A readme or some notes on the PR for now will do.

Added documentation to the README

**Comment by Swiftyos** [comment]
> Also getting the following error after the `test-storybook` run which causes a non zero exit. I bet we are missing a config somewhere. https://storybook.js.org/docs/writing-tests/snapshot-testing/snapshot-testing#override-the-default-snapshot-directory
> 
> ```
> ..... <more tests passing here>
>  PASS   browser: chromium  src/components/ui/switch.stories.tsx
> 
> Snapshot Summary
>  › 6 snapshot files obsolete from 6 test suites. To remove them all, re-run jest with `-u`.
>    ↳   • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-6163898.snap
>        • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-6166598.snap
>        • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-6166758.snap
>        • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-6173890.snap
>        • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-61738C8.snap
>        • autogpt_platform/supabase/docker/volumes/db/data/pg_logical/snapshots/0-6173978.snap
> 
> Test Suites: 33 passed, 33 total
> Tests:       155 passed, 155 total
> Snapshots:   6 files obsolete, 0 total
> Time:        18.452 s
> Ran all test suites.
> error Command failed with exit code 1.
> ```

Thanks, I've fixed this by adding jest config

**Comment by ntindle** [comment]
> > The fonts don't seem to be loading. Not sure why as I only looked at this tutorial https://stackonfire.com/posts/shadcn-storybook-visual-testing/ but here's a photo for reference image
> 
> Font Family Times is the default font for the project so, whilst it does look off it is working as expected.

Wait are all our buttons and such styled like that? That’s horrid

**Comment by Swiftyos** [comment]
> Wait are all our buttons and such styled like that? That’s horrid

These are the base components in components/ui that I believe have been copied over from shadcn or radix ui. They are then further tweaked when used. 

I'm going to make agptui dir in the components dir to include our own styled components based on the new designs

**Review by ntindle — APPROVED** [review]

File: autogpt_platform/frontend/package.json
Comment (Pwuts) [inline autogpt_platform/frontend/package.json:1]: This was merged without updating `yarn.lock`. This breaks the docker build, which requires `yarn.lock` to be consistent with `package.json`.
